packages/server/src/connection-server.ts

```diff
@@ -1,71 +1,46 @@
-import { Operation, fork } from 'effection';
+import { Operation, spawn } from 'effection';
+import { subscribe, ChainableSubscription } from '@effection/subscription';
 import { Mailbox } from '@bigtest/effection';
 import { Atom } from '@bigtest/atom';
 import { OrchestratorState } from './orchestrator/state';
-import { express, Socket } from '@bigtest/effection-express';
+import { AgentConnection, createAgentHandler, Command, TestEvent } from '@bigtest/agent';
 
 interface ConnectionServerOptions {
-  inbox: Mailbox;
+  inbox: Mailbox<Command>;
   delegate: Mailbox;
   atom: Atom<OrchestratorState>;
   port: number;
   proxyPort: number;
   manifestPort: number;
 };
 
-let counter = 1;
-
-export function generateAgentId(): string {
-  return `agent.${counter++}`;
-}
-
 export function* createConnectionServer(options: ConnectionServerOptions): Operation {
-  function* handleConnection(socket: Socket): Operation {
-    console.debug('[connection] connected');
-
-    let messages: Mailbox = yield socket.subscribe();
+  let handler: ChainableSubscription<AgentConnection, void> = yield createAgentHandler(options.port);
 
-    let { data, agentId } = yield messages.receive({ type: 'connected' });
-
-    agentId = agentId || generateAgentId();
-
-    let agent = options.atom.slice('agents', agentId);
+  options.delegate.send({ status: "ready" });
 
-    try {
-      console.log(`[connection] connected ${agentId}`);
+  while(true) {
+    let connection: AgentConnection = yield handler.expect();
+    yield spawn(function*() {
+      console.log(`[connection] connected ${connection.agentId}`);
+      let agent = options.atom.slice('agents', connection.agentId);
 
-      agent.set({ ...data, agentId });
+      agent.set({ ...connection.data, agentId: connection.agentId });
 
-      yield fork(function*() {
+      yield spawn(function*(): Operation<void> {
         while (true) {
-          console.debug('[connection] waiting for message', agentId);
-          let message = yield options.inbox.receive({ agentId: agentId });
-
-          yield socket.send(message);
+          let message = yield options.inbox.receive({ agentId: connection.agentId });
```

Comment:
We could take this even further here and make the inbox a Channel, but I wanted to leave this to a future PR.